include/engine/routing_algorithms/routing_base.hpp

```diff
@@ -393,7 +398,9 @@ template <class DataFacadeT, class Derived> class BasicRoutingInterface
             BOOST_ASSERT(i < id_vector.size());
             BOOST_ASSERT(phantom_node_pair.target_phantom.forward_travel_mode > 0);
             unpacked_path.emplace_back(
-                PathData{id_vector[i], phantom_node_pair.target_phantom.name_id, weight_vector[i],
+                PathData{id_vector[i],
+                         phantom_node_pair.target_phantom.name_id,
+                         weight_vector[i-delta],
```

Comment:
If we emplace back we don't need the `PathData` constructor. In case this was the situation where emplace_back was failing without the constructor, use push_back, as there is an rvalue overload.